src/Agent/NewRelic/Agent/Core/Api/AgentApi.cs

```diff
@@ -468,7 +469,31 @@ public static string GetBrowserTimingHeader()
         {
             const ApiMethod apiMetric = ApiMethod.GetBrowserTimingHeader;
             const string apiName = nameof(GetBrowserTimingHeader);
-            return TryInvoke(InternalApi.GetBrowserTimingHeader, apiName, apiMetric) ?? string.Empty;
+            return TryInvoke(InternalApi.GetBrowserTimingHeader, null, apiName, apiMetric) ?? "";
+        }
+
+        /// <summary>
+        /// Returns the html snippet to be inserted into the header of html pages to enable Real User Monitoring.
+        /// The html will instruct the browser to fetch a small JavaScript file and start the page timer.
+        /// Supports web applications only.
+        /// </summary>
+        /// <param name="nonce">An optional per-request, cryptographic nonce used by a <c>Content-Security-Policy</c> <c>script-src</c> policy.</param>
+        /// <example>
+        /// <code>
+        /// &lt;html>
+        ///   &lt;head>
+        ///     &lt;&#37;= NewRelic.Api.Agent.NewRelic.GetBrowserTimingHeader()&#37;>
+        ///   &lt;/head>
+        ///   &lt;body>
+        ///   ...
+        /// </code>
+        /// </example>
+        /// <returns>An html string to be embedded in a page header.</returns>
+        public static string GetBrowserTimingHeader(string nonce)
+        {
+            const ApiMethod apiMetric = ApiMethod.GetBrowserTimingHeader;
+            const string apiName = nameof(GetBrowserTimingHeader);
+            return TryInvoke(InternalApi.GetBrowserTimingHeader, nonce, apiName, apiMetric) ?? "";
```

Comment:
Instead of adding `parameter` to `TryInvoke` for this functionality, why not do the same using closures?
`TryInvoke(() => InternalApi.GetBrowserTimingHeader(nonce), apiName, apiMetric)`

small nit- `""` is inconsistent with `string.Empty` in the rest of the file.

Reply:
It was to avoid creating another closure and another allocation... but maybe it would be the same either way? I'd have to look at the IL or the output from a memory profiler.

But good point; it'd be simpler that way.

Reply:
This method:

```
    IL_0000: ldnull
    IL_0001: ldftn        string [NewRelic.Agent.Extensions]NewRelic.Agent.Api.InternalApi::GetBrowserTimingHeader(string)
    IL_0007: newobj       instance void class [mscorlib]System.Func`2<string, string>::.ctor(object, native int)
    IL_000c: ldnull
    IL_000d: ldstr        "GetBrowserTimingHeader"
    IL_0012: ldc.i4.5
    IL_0013: call         !!0/*string*/ NewRelic.Agent.Core.AgentApi::TryInvoke<string>(class [mscorlib]System.Func`2<string, !!0/*string*/>, string, string, valuetype NewRelic.Agent.Core.Metric.ApiMethod)
    IL_0018: dup
    IL_0019: brtrue.s     IL_0021
    IL_001b: pop
    IL_001c: ldstr        ""
    IL_0021: ret
```

Proposed closure.

```
  **IL_0000: newobj       instance void NewRelic.Agent.Core.AgentApi/'<>c__DisplayClass27_0'::.ctor()
    IL_0005: dup
    IL_0006: ldarg.0      // nonce
    IL_0007: stfld        string NewRelic.Agent.Core.AgentApi/'<>c__DisplayClass27_0'::nonce
    IL_000c: ldftn        instance string NewRelic.Agent.Core.AgentApi/'<>c__DisplayClass27_0'::'<GetBrowserTimingHeader>b__0'()
  IL_0012: newobj       instance void class [mscorlib]System.Func`1<string>::.ctor(object, native int)
    IL_0017: ldstr        "GetBrowserTimingHeader"
    IL_001c: ldc.i4.5
    IL_001d: call         !!0/*string*/ NewRelic.Agent.Core.AgentApi::TryInvoke<string>(class [mscorlib]System.Func`1<!!0/*string*/>, string, valuetype NewRelic.Agent.Core.Metric.ApiMethod)
    IL_0022: dup
    IL_0023: brtrue.s     IL_002b
    IL_0025: pop
    IL_0026: ldstr        ""
    IL_002b: ret
```

Note the additional allocation for the closure. This would happen per-request, so I thought it might be worth avoiding.

OTOH, allocations don't appear to be highly optimised in the current code (we could get rid of allocating the `Func` per request if we were worried about that, for example), so perhaps the simpler code is better.

Reply:
Interesting question. It would be cool to see what kind of empirical impact it has. I imagine it might not even be detectable without extreme precision.